client/state/support/actions.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SUPPORT_USER_ACTIVATED,
+	SUPPORT_USER_DEACTIVATED,
+	SUPPORT_USER_FETCH_TOKEN,
+} from 'state/action-types';
+
+import User from 'lib/user';
+
+import userSettings from 'lib/user-settings';
+
+/**
+ * Requests a support user token, then dispatches the relevant actions upon response
+ * 
+ * @param  {string} supportUser     Support username
+ * @param  {string} supportPassword Support password
+ * @return {thunk}                  The action thunk
+ */
+export function supportUserFetchToken( supportUser, supportPassword ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SUPPORT_USER_FETCH_TOKEN,
+			supportUser
+		} );
+
+		if ( supportUser && supportPassword ) {
```

Comment:
this is another place to use early-abort instead of nesting. is there a reason we would dispatch the above action if `supportUser` were missing?

```js
return dispatch => {
	if ( ! supportUser || ! supportPassword ) {
		return;
	}

	dispatch( { type: SUPPORT_USER_FETCH_TOKEN, supportUser } );

	const user = new User();
	
	user.changeUser( … );
}
```